### PR TITLE
[REFACTORING] Cleanup a bit mailbox-jpa & POJOify Property

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
test this please

```
[f233b3fa6fcbb975315f9defcc7e565be571d57e] [ERROR] Failures: 
[f233b3fa6fcbb975315f9defcc7e565be571d57e] [ERROR]   EventDeadLettersIntegrationTest.groupIdOfFailedEventShouldBeStoredInDeadLetter:240 1 expectation failed.
[f233b3fa6fcbb975315f9defcc7e565be571d57e] JSON path . doesn't match.
[f233b3fa6fcbb975315f9defcc7e565be571d57e] Expected: iterable with items ["org.apache.james.webadmin.integration.EventDeadLettersIntegrationTest$RetryEventsListenerGroup"] in any order
[f233b3fa6fcbb975315f9defcc7e565be571d57e]   Actual: []
```

**Comment by chibenwa** [comment]
```
[f233b3fa6fcbb975315f9defcc7e565be571d57e] [ERROR] Failures: 
[f233b3fa6fcbb975315f9defcc7e565be571d57e] [ERROR]   EventDeadLettersIntegrationTest.groupIdOfFailedEventShouldBeStoredInDeadLetter:240 1 expectation failed.
[f233b3fa6fcbb975315f9defcc7e565be571d57e] JSON path . doesn't match.
[f233b3fa6fcbb975315f9defcc7e565be571d57e] Expected: iterable with items ["org.apache.james.webadmin.integration.EventDeadLettersIntegrationTest$RetryEventsListenerGroup"] in any order
[f233b3fa6fcbb975315f9defcc7e565be571d57e]   Actual: []
```

test this please

**Comment by chibenwa** [comment]
Was :green_apple: , rebased to solve simple conflicts.

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageDAO.java [hunk 1]
```diff
@@ -278,7 +278,7 @@ private UDTValue toUDT(MessageAttachment messageAttachment) {
     private PropertyBuilder getPropertyBuilder(Row row) {
         PropertyBuilder property = new PropertyBuilder(
             row.getList(PROPERTIES, UDTValue.class).stream()
-                .map(x -> new SimpleProperty(x.getString(Properties.NAMESPACE), x.getString(Properties.NAME), x.getString(Properties.VALUE)))
+                .map(x -> new Property(x.getString(Properties.NAMESPACE), x.getString(Properties.NAME), x.getString(Properties.VALUE)))
```
Comment (remk) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageDAO.java]: Not this PR but `x` ?

Comment (chibenwa) [reply mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageDAO.java]: Sounds like a silly name, yes!

**Comment by chibenwa** [comment]
```
"message": "Server Error"

Caused by: java.io.IOException: Server returned HTTP response code: 502 for URL: https://api.github.com/repos/linagora/james-project
```

test this please

**Comment by Arsnael** [comment]
```
[efaa2e2deb0d281bce273528f1b3d65d53ff7434] [ERROR] Errors: 
[efaa2e2deb0d281bce273528f1b3d65d53ff7434] [ERROR]   CassandraMailRepositoryTest.storingAndRemovingMessagesConcurrentlyShouldLeadToConsistentResult » Execution
[efaa2e2deb0d281bce273528f1b3d65d53ff7434] [ERROR]   CassandraMailRepositoryTest.storingAndRemovingMessagesConcurrentlyShouldLeadToConsistentResult » Execution
[efaa2e2deb0d281bce273528f1b3d65d53ff7434] [ERROR]   CassandraMailRepositoryTest.storingAndRemovingMessagesConcurrentlyShouldLeadToConsistentResult » Execution
[efaa2e2deb0d281bce273528f1b3d65d53ff7434] [ERROR]   CassandraMailRepositoryTest.storingAndRemovingMessagesConcurrentlyShouldLeadToConsistentResult » Execution
[efaa2e2deb0d281bce273528f1b3d65d53ff7434] [ERROR]   CassandraMailRepositoryTest.storingAndRemovingMessagesConcurrentlyShouldLeadToConsistentResult » Execution
[efaa2e2deb0d281bce273528f1b3d65d53ff7434] [ERROR]   CassandraMailRepositoryTest.storingAndRemovingMessagesConcurrentlyShouldLeadToConsistentResult » Execution
[efaa2e2deb0d281bce273528f1b3d65d53ff7434] [ERROR]   CassandraMailRepositoryTest.storingAndRemovingMessagesConcurrentlyShouldLeadToConsistentResult » Execution
```
those are being too recurrent... 

test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**